common-pg/src/main/scala/com/socrata/pg/soql/SoQLAnalysisSqlizer.scala

```diff
@@ -1,142 +1,282 @@
+      case analysis: SoQLAnalysis[UserColumnId, SoQLType]  =>
+        set + analysis
+    }
+  }
+}
+
+trait SoQLAnalysisSqlizer {
+  this: Sqlizer[(BinaryTree[SoQLAnalysis[UserColumnId, SoQLType]], Map[TableName, String], Seq[SqlColumnRep[SoQLType, SoQLValue]])] =>
+
+  import Sqlizer._
+  import SqlizerContext._
+
+  /**
+   * Convert one analysis to parameteric sql.
+   */
+  def toSql(analysis: SoQLAnalysis[UserColumnId, SoQLType], // scalastyle:ignore method.length parameter.number
+            prevAna: Option[SoQLAnalysis[UserColumnId, SoQLType]],
+            tableNames: Map[TableName, String],
+            allColumnReps: Seq[SqlColumnRep[SoQLType, SoQLValue]],
+            reqRowCount: Boolean,
+            rep: Map[QualifiedUserColumnId, SqlColumnRep[SoQLType, SoQLValue]],
+            typeRep: Map[SoQLType, SqlColumnRep[SoQLType, SoQLValue]],
+            setParams: Seq[SetParam],
+            context: Context,
+            escape: Escape,
+            fromTableName: Option[TableName] = None): (ParametricSql, /* params count in select, excluding where, group by... */ Int) = {
 
     // Use leading search despite being poor in semantics.  It is more likely to use the GIN index and performs better.
     // TODO: switch to trailing search when there is smarter index support
     val leadingSearch = context.getOrElse(LeadingSearch, true) != false
 
     val ana = if (reqRowCount) rowCountAnalysis(analysis) else analysis
-    val ctx = context + (Analysis -> analysis) +
-                        (TableMap -> tableNames) +
-                        (TableAliasMap -> tableNames.map { case (k, v) => (k.qualifier, realAlias(k, v)) })
 
     val joins = analysis.joins
 
+    val fromTableNames = ana.from.foldLeft(Map.empty[TableName, String]) { (acc, from) =>
+      acc + (from.copy(alias = None) -> from.name)
+    }
+    val fromTableAliases = ana.from.foldLeft(Map.empty[String, String]) { (acc, from) =>
+      from.alias match {
+        case Some(alias) =>
+          acc + (alias -> from.name)
+        case None =>
+          acc
+      }
+    }
+
+    val simpleJoinMapAcc: Map[String, String] = analysis.from match {
+      case Some(TableName(name, Some(alias))) =>
+        Map(alias -> name, name -> alias)
+      case Some(TableName(name, None)) =>
+        Map(name -> name)
+      case _ =>
+        Map.empty
+    }
+
+    val simpleJoinMap = joins.foldLeft(simpleJoinMapAcc) { (acc, j) =>
+      j.from.subAnalysis match {
+        case Left(TableName(name, Some(alias))) =>
+          acc + (alias -> name) + (name -> alias)
+        case Left(TableName(name, None)) =>
+          // simple join w/o alias does not need remap
+          acc + (name -> name)
+        case _ =>
+          // complex join reps are just qualifier.name and use type reps and don't need entries in reps.  They use
+          acc
+      }
+    }
+
+    val ctx = context + (Analysis -> analysis) +
+      (SimpleJoinMap -> simpleJoinMap) +
```

Comment:
Are we adding all the forms of joins for VQE here?

Reply:
There are two types of joins - complex like "join (select...)" and simple join view.  We only need a map of simple joins.